cargo-dist/src/platform.rs

```diff
@@ -417,3 +436,108 @@ fn supports(
     }
     res
 }
+
+impl RuntimeConditions {
+    fn merge(&mut self, other: &Self) {
+        let RuntimeConditions {
+            min_glibc_version,
+            min_musl_version,
+            rosetta2,
+        } = other;
+
+        self.min_glibc_version =
+            max_of_min_libc_versions(&self.min_glibc_version, min_glibc_version);
+        self.min_musl_version = max_of_min_libc_versions(&self.min_musl_version, min_musl_version);
+        self.rosetta2 |= rosetta2;
+    }
+}
+
+/// Combine two min_libc_versions to get a new min that satisfies both
+fn max_of_min_libc_versions(
+    lhs: &Option<LibcVersion>,
+    rhs: &Option<LibcVersion>,
+) -> Option<LibcVersion> {
+    match (*lhs, *rhs) {
+        (None, None) => None,
+        (Some(ver), None) | (None, Some(ver)) => Some(ver),
+        (Some(lhs), Some(rhs)) => Some(lhs.max(rhs)),
```

Comment:
Does the default `ord` derive handle this for us or do we have to implement our own so it knows how to compare both values?